subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -31,26 +38,38 @@ static const char *const sensor_type_str[] = {
 	[NRF_CLOUD_LTE_LINK_RSRP] = "RSRP",
 	[NRF_CLOUD_DEVICE_INFO] = "DEVICE",
 };
+#endif
 
-/* --- A few wrappers for cJSON APIs --- */
-
-static int json_add_obj(cJSON *parent, const char *str, cJSON *item)
+int nrf_cloud_codec_init(void)
 {
-	cJSON_AddItemToObject(parent, str, item);
+	cJSON_Init();
 
 	return 0;
 }
 
-static int json_add_str(cJSON *parent, const char *str, const char *item)
+static char *json_strdup(cJSON * const string_obj)
 {
-	cJSON *json_str;
+	char *dest;
+	char *src = cJSON_GetStringValue(string_obj);
 
-	json_str = cJSON_CreateString(item);
-	if (json_str == NULL) {
-		return -ENOMEM;
+	if (!src) {
+		return NULL;
 	}
 
-	return json_add_obj(parent, str, json_str);
+	dest = nrf_cloud_calloc(strlen(src) + 1, 1);
+	if (dest) {
+		strcpy(dest, src);
+	}
+
+	return dest;
+}
+
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+static int json_add_obj(cJSON *parent, const char *str, cJSON *item)
+{
+	cJSON_AddItemToObject(parent, str, item);
```

Comment:
`cJSON_AddItemToObject()` can fail.  Any reason to suppress failures here?

Reply:
this is an existing function:
https://github.com/nrfconnect/sdk-nrf/blob/7db61c9ac892d621decf6f2e695444ba8e6100e4/subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c#L37

Reply:
updated to handle "new" `cJSON_AddItemToObject` return value